src/renderer/dx/DxFontRenderData.cpp

```diff
@@ -118,6 +118,14 @@ DxFontRenderData::DxFontRenderData(::Microsoft::WRL::ComPtr<IDWriteFactory1> dwr
                                                                                                   DWRITE_FONT_STYLE style,
                                                                                                   DWRITE_FONT_STRETCH stretch)
 {
+    if (weight == _defaultFontInfo.GetWeight() &&
+        style == _defaultFontInfo.GetStyle() &&
+        stretch == _defaultFontInfo.GetStretch() &&
+        _textFormatMap.find(_defaultFontInfo) != _textFormatMap.end())

```

Comment:
The lookup here is expensive because of how `std::hash<DxFontInfo>` is implemented.

Reply:
Does the lookup become cheap if you fix `std::hash<DxFontInfo>`?

Reply:
Couple things, though:

You're using `find` _and_ `at`, which causes two lookups. You should cache the result of `find` and then use the iterator to get the return value.

I don't understand what this is doing. Is this a comparison that ignores the font name?

If so, this should be a member of DxFontInfo. If not, we need to explain what it is.

If we never want to use the font family name, it should not be part of the hash...?

Reply:
> Is this a comparison that ignores the font name?

Yes, it is. Because the only way that can change the font name is through `UpdateFont`. After `UpdateFont`, all the lookup focus on the 3 font attributes. 

> If we never want to use the font family name, it should not be part of the hash...?

In my last `DxFontInfo` PR, intuitively I wanted everything inside `DxFontInfo` to be part of the hash. But in this PR I realized that all the actual lookup operation does NOT need font family name, because of the reason I mention above.

Then again, it feels weird to me to not include font name in the hash. What if we need to use the hash somewhere else where the family name *does* matter? I'm a bit lost here.